### DISCUSSION TITLE
Can I capture raw icmp packet by libnet?

### DISCUSSION THREAD

**Discussion by RyanCode** [libnet/libnet#155] [Q&A] [answered]
The Linux kernel filters out IP packets at Layer 3 whose destination address is not local，Does libnet provide icmp capability to capture non native ip?

**Comment by ivalery111** [comment, marked as answer]
Hey,

As I know, a libnet is about a _crafting_ the network packets. You want to _capture_ the packets. This can be done with pcap or another tools.

IMHO, [this](http://www.cas.mcmaster.ca/~rzheng/course/COSC6397sp2008/Libpcap_libnet.pdf) example will be useful for you.
